src/modules/launcher/Plugins/Microsoft.Plugin.Uri/Main.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Microsoft.Plugin.Uri.UriHelper;
+using Wox.Infrastructure.Logger;
+using Wox.Infrastructure.Storage;
+using Wox.Plugin;
+
+namespace Microsoft.Plugin.Uri
+{
+    public class Main : IPlugin, IPluginI18n, IContextMenu, ISavable, IDisposable
+    {
+        private readonly ExtendedUriParser _uriParser;
+        private readonly UriResolver _uriResolver;
+        private readonly PluginJsonStorage<UriSettings> _storage;
+        private bool _disposed;
+        private UriSettings _uriSettings;
+        private RegisteryWrapper _registeryWrapper;
+
+        public Main()
+        {
+            _storage = new PluginJsonStorage<UriSettings>();
+            _uriSettings = _storage.Load();
+            _uriParser = new ExtendedUriParser();
+            _uriResolver = new UriResolver();
+            _registeryWrapper = new RegisteryWrapper();
+        }
+
+        public string BrowserIconPath { get; set; }
+
+        public string DefaultIconPath { get; set; }
+
+        public PluginInitContext Context { get; protected set; }
+
+        public List<ContextMenuResult> LoadContextMenus(Result selectedResult)
+        {
+            return new List<ContextMenuResult>(0);
+        }
+
+        public List<Result> Query(Query query)
+        {
+            var results = new List<Result>();
+
+            if (!string.IsNullOrEmpty(query?.Search)
+                && _uriParser.TryParse(query.Search, out var uriResult)
+                && _uriResolver.IsValidHost(uriResult))
+            {
+                var uriResultString = uriResult.ToString();
+
+                results.Add(new Result
+                {
+                    Title = uriResultString,
+                    SubTitle = Context.API.GetTranslation("Microsoft_plugin_uri_website"),
+                    IcoPath = _uriSettings.ShowBrowserIcon
+                        ? BrowserIconPath
+                        : DefaultIconPath,
+                    Action = action =>
+                    {
+                        Process.Start(new ProcessStartInfo(uriResultString)
+                        {
+                            UseShellExecute = true,
+                        });
+                        return true;
+                    },
+                });
+            }
+
+            return results;
+        }
+
+        public void Init(PluginInitContext context)
+        {
+            Context = context ?? throw new InvalidOperationException($"{nameof(Context)} should not be NULL");
```

Comment:
nit:  I think `ArgumentNullException` is more appropriate here.

Reply:
I didn't want to throw a NullReferenceException (As per Language spec), this one is indeed more appropriate.